### PR TITLE
Migrate from VITE to NEXTjs

### PR DESCRIPTION
**Pull request by yamin-faruque-sazim**
**Description of Changes**

- Installed nextjs and necessary dependencies 
- Updated file folder structure and imports 
- Created new directories (pages, modules, shared)
- Updated css file to css modules to be used locally 
- Updated fixed local storage logic to properly use useffect to load state from local storage
- Added logic to local storage so that it checks if it's being run on the client side and not in the server side during initial request
- Removed unnecessary files and folders

### CODE REVIEW

File: modules/HomePage/components/TaskList/TaskList.tsx [hunk 1]
```diff
@@ -10,7 +10,7 @@ import {
   Select,
 } from "@mantine/core";
 import { useDisclosure } from "@mantine/hooks";
-import { Task } from "../../../types/Task";
+import { Task } from "../../types/Task";
```
Comment (ishmam-iqbal-sazim) [inline modules/HomePage/components/TaskList/TaskList.tsx:15]: Should be a line gap between relative and library imports

File: pages/_document.page.tsx [hunk 2]
```diff
@@ -0,0 +1,18 @@
+
+import Document, { Html, Head, Main, NextScript } from 'next/document';
```
Comment (ishmam-iqbal-sazim) [inline pages/_document.page.tsx]:
```suggestion
import Document, { Html, Head, Main, NextScript } from "next/document";
```
Can take some time out to configure prettier such that all single quotes are changed to double quotes on save.

File: modules/HomePage/containers/MainContainer.tsx [hunk 3]
```diff
@@ -1,30 +1,39 @@
...
+          dueDate: new Date(task.dueDate),
+        }))
+      );
+    }
+  }, []);
+
+  // Load tasks from localStorage on the client side only
+ // Empty dependency array, runs only once on component mount
+
+  // Save tasks to localStorage whenever tasks state changes
```
Comment (ishmam-iqbal-sazim) [inline modules/HomePage/containers/MainContainer.tsx]: Should remove the comments.

**Review by ishmam-iqbal-sazim — APPROVED** [review]
Approved pending comments.

File: pages/_app.page.tsx [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+import "@mantine/dates/styles.css";
+import { MantineProvider } from "@mantine/core";
+import type { AppProps } from "next/app";
+
+
+
+export default function App({ Component, pageProps }: AppProps) {
+  return (
+    <MantineProvider defaultColorScheme="dark">
+
```
Comment (ragib-hossain-sazim) [inline pages/_app.page.tsx]: [Nitpick] Unnecessary new line

**Review by ragib-hossain-sazim — APPROVED** [review]
Approved pending comments.

**Comment by yamin-faruque-sazim** [comment]
Addressed all the comments

**Pull request merged**